### ISSUE TITLE
Slow datashader.regrid when panning around a dask array

### ISSUE THREAD

**Issue by eddienko** [holoviz/holoviews#4435]
I have been observing a low performance when using datashader.regrid and panning around a dask array. A code snippet that shows the problem is e.g.:

```
import dask.array as da
import holoviews as hv
import holoviews.operation.datashader as hd

hv.extension('bokeh')

r = da.random.random((5000,5000))

hd.regrid(hv.Image((range(5000), range(5000), r)))
```

The following GIF shows the issue, it is not too bad here, but in a real example with large images stored on disk it is really noticeable.

![before](https://user-images.githubusercontent.com/3780274/82759809-7be78c80-9de7-11ea-9793-6427df4e77e9.gif)

I have been looking at a workaround, and this is my take modifying `holoviews/operation/datasahder.py`

```
--- datashader.py.orig  2020-05-24 16:58:28.016298741 +0000
+++ datashader.py       2020-05-24 16:59:20.510295533 +0000
@@ -847,6 +847,13 @@
         info = self._get_sampling(element, x, y)
         (x_range, y_range), (xs, ys), (width, height), (xtype, ytype) = info
 
+        # Make panning smoother
+        xdiff = (x_range[1] - x_range[0]) / 2
+        ydiff = (y_range[1] - y_range[0]) / 2
+
+        x_range = (max(0, x_range[0]-xdiff), min(x_range[1]+xdiff, coords[0].max()))
+        y_range = (max(0, y_range[0]-ydiff), min(y_range[1]+ydiff, coords[1].max()))
+
         # Disable upsampling by clipping size and ranges
         (xstart, xend), (ystart, yend) = (x_range, y_range)
         xspan, yspan = (xend-xstart), (yend-ystart)
```

Basically I am extending a little bit the range of the regridded image outside of the visible area. The result is quite better:

![after2](https://user-images.githubusercontent.com/3780274/82760259-527c3000-9dea-11ea-9e24-06ccd6873c81.gif)

I am not sure however if this is the correct approach or will break anything. So looking for advice, before submitting a PR (if it looks ok).

Using holoviews 1.13.2.

**Comment by philippjfr** [comment]
Thanks for the issue. This seems like a nice improvement but in the end it's not really making it smoother, it just computes some buffer and makes it **appear** smoother. I'd be happy to have an option for this but I don't think it should be the default since it "wastes" resolution on areas that aren't visible and isn't accurately returning the ranges that are being requested. I'd suggest a configurable padding or buffer value as a fraction of the overall span. Ideally the width/height would also get adjusted to account for the additional buffer area.

**Comment by jbednar** [comment]
I agree; this is nice, but should be configurable, and probably should be zero buffer by default, because panning is only one of the many things that get done with a datashader figure.  Slowing things down for all cases by rendering unseen data will only be appropriate for the specific case of someone expecting to pan around (as opposed to zooming in or the much more common just look at it without interacting with it), so it should be enabled as needed.  But in that case, yes, it will improve the experience, so I'd be happy for this to be a supported option. Thanks!
